pkg/endpoint/policy.go

```diff
@@ -286,6 +286,9 @@ func (e *Endpoint) SetIdentity(owner Owner, id *policy.Identity) {
 	defer tree.Mutex.Unlock()
 	cache := owner.GetConsumableCache()
 
+	e.SecLabel = id
+	e.Consumable = cache.GetOrCreate(id.ID, id)
+
```

Comment:
Right, let's just do `cache.Remove() if e.Consumable != nil` first and then `GetOrCreate`